src/empire/object.c

```diff
@@ -93,13 +93,20 @@ void empire_object_load(buffer *buf, int version)
             for (int r = 0; r < 8; r++) {
                 old_buys_resource[r] = buffer_read_u8(buf);
             }
-        } else {
+        } else if (version <= SCENARIO_LAST_EMPIRE_RESOURCES_U8) {
             for (int r = RESOURCE_MIN; r < RESOURCE_MAX; r++) {
                 full->city_sells_resource[r] = buffer_read_u8(buf);
             }
             for (int r = RESOURCE_MIN; r < RESOURCE_MAX; r++) {
                 full->city_buys_resource[r] = buffer_read_u8(buf);
             }
+        } else {
+            for (int r = RESOURCE_MIN; r < RESOURCE_MAX; r++) {
+                full->city_sells_resource[r] = buffer_read_i32(buf);
```

Comment:
I agree an `u8` set a small limit, but why not `u16`?

This is wasting 45 bytes per object down below.

Considering the price per gigabyte of the newest SSD's, that a waste of a perfectly good $0.00000001 dollars. Unacceptable!

Reply:
Riiiight 😛 I think I’d like to keep it as an int in case we want to use negatives to flag special values in the future.

Reply:
You can use `i16` though, I mean I doubt the max amount will ever be 32768. You don't even have to bump the version for that as this is an internal change in an in-progress PR  😉

Reply:
Actually Zardo has been developing a map with early builds of this PR and I’d prefer not to break what he’s done so far. Thinking about it more, a better approach might be to skip serializing resources entirely unless it’s a city.

Reply:
Fair enough. I'll try to handle my OCD then 😉